From 47f19fac2fdbbf5c5b5124883043e473f8c44edd Mon Sep 17 00:00:00 2001
From: Robby Stephenson <robby@periapsis.org>
Date: Tue, 2 Oct 2018 21:02:37 -0400
Subject: Fix bug when using regexp filter for number fields

Only use a QIntValidator when the filter rule is not a regular
expression.

BUG: 399323
FIXED-IN: 3.1.4
FIXED-IN: 3.2
---
 ChangeLog                    | 4 ++++
 src/gui/filterrulewidget.cpp | 3 ++-
 2 files changed, 6 insertions(+), 1 deletion(-)

diff --git a/ChangeLog b/ChangeLog
index 879f4d0..3284dc7 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,3 +1,7 @@
+2018-10-02  Robby Stephenson  <robby@periapsis.org>
+
+	* Fixed bug when using a regular expression for filtering in a number field (Bug 399323).
+
 2018-09-02  Robby Stephenson  <robby@periapsis.org>
 
 	* Released Tellico 3.1.3.
diff --git a/src/gui/filterrulewidget.cpp b/src/gui/filterrulewidget.cpp
index 8e3d7b1..aa3718d 100644
--- a/src/gui/filterrulewidget.cpp
+++ b/src/gui/filterrulewidget.cpp
@@ -169,7 +169,8 @@ void FilterRuleWidget::slotRuleFunctionChanged(int which_) {
   } else {
     m_valueStack->setCurrentWidget(m_ruleValue);
     m_ruleValue->setPlaceholderText(QString());
-    if(m_ruleType == Number) {
+    if(m_ruleType == Number &&
+      (data != FilterRule::FuncRegExp && data != FilterRule::FuncNotRegExp)) {
       m_ruleValue->setValidator(new QIntValidator(this));
     } else {
       m_ruleValue->setValidator(nullptr);
-- 
cgit v0.11.2
